compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -2015,3 +2015,6 @@ error.inferred.query.construct.type.as.stream=\
 
 error.invalid.binding.pattern.in.on.fail=\
   invalid binding pattern in ''on fail'' clause: only a capture binding pattern or an error binding pattern is allowed
+
+error.undefined.constant.symbol=\
+  undefined constant symbol ''{0}''
```

Comment:
Is it necessary to define a new error?

Reply:
I guess yes. Similar errors related to this issue include `undefined symbol` and `expression is not a constant expression`. There are two distinct scenarios to address:

1. In the first case, there exists a symbol in the scope, but it is not a constant.
2. In the second case, there is no symbol in the scope.

Using the `undefined symbol` error would be inaccurate in the first case. On the other hand, employing the `expression is not a constant expression` error is not feasible as it includes a broader range of expressions.
[const-expr defined in the spec](https://ballerina.io/spec/lang/master/#section_6.4)

Reply:
Variables and constants share the [same symbol space](https://ballerina.io/spec/lang/master/#section_5.9.3), and therefore, using a single `undefined symbol` error is OK for when a symbol is undefined, IMO.

Also, the requirements for constant expressions are specific, and I believe it is better to give the exact error.

Reply:
ACK. Let's keep a new error for this.

Reply:
> Variables and constants share the [same symbol space](https://ballerina.io/spec/lang/master/#section_5.9.3), and therefore, using a single `undefined symbol` error is OK for when a symbol is undefined, IMO.
> 
> Also, the requirements for constant expressions are specific, and I believe it is better to give the exact error.

+1, IMO this error msg is not correct. We should show a better one that says the expression isn't a constant. By seeing this error message, if someone tries to declare another constant with the same variable, they get an error saying the symbol is redeclared.